parsel/csstranslator.py

```diff
@@ -89,6 +101,116 @@ def xpath_text_simple_pseudo_element(self, xpath):
         """Support selecting text nodes using ::text pseudo-element"""
         return XPathExpr.from_xpath(xpath, textnode=True)
 
+    def xpath_first_of_pseudo(self, xpath):
```

Comment:
Hey @redapple -- I'm just seeking to understand here (not judging), I'm still a noob when it comes to cssselect, I believe you know better than me if it should be implemented differently.
This is cool, I didn't knew cssselect already provided those kind of hooks. =)

Thinking about these extensions, I think it would be nice if it matched what someone that already knows CSS would expect -- so, the closer to it, the better.

Shouldn't it be `nth-of-type` and `first-of-type` instead of `nth-of` and `first-of`?